Makefile

```diff
@@ -322,33 +316,18 @@ switch-eks:
 ##  --    Docker images    --  ##
 #################################
 
+docker-login:
+	@ docker login -u $(ELASTIC_DOCKER_LOGIN) -p $(ELASTIC_DOCKER_PASSWORD) push.docker.elastic.co 2> /dev/null
```

Comment:
For local dev use: how is the `ELASTIC_DOCKER_PASSWORD` being set up? If this is a manual step each dev has do upfront we should document it. But I wonder if we can have something automated?

Reply:
Good point. `ELASTIC_DOCKER_PASSWORD` is defined in the file `.registry.env` 
when `make -C .ci get-docker-creds` is executed.

It's fairly easy to automate:
```diff
diff --git a/Makefile b/Makefile
index b6356b8e..5cc5fc63 100644
--- a/Makefile
+++ b/Makefile
@@ -316,7 +316,10 @@ switch-eks:
 ##  --    Docker images    --  ##
 #################################
 
-docker-login:
+.registry.env:
+       make -sC .ci get-docker-creds
+
+docker-login: .registry.env
```

Reply:
Otherwise, another way of looking at that is to keep the registry configurable to be able to easily use at least GCE as today for dev needs.

I seem to introduce a kind of regression by removing this feature.

So far I did not manage a nice way to do this by reconciling simplicity given the specifics of each registry, which is the purpose of this task: simplify.
- the authentication is different: docker login or gcloud auth
- Elastic registry needs a retag with the push prefix
- different namespaces for the images (eck, eck-snapshots, eck-ci) vs gcloud_project for GCE

I might have something reasonable this morning.

Reply:
Would it make sense to have a `.sentenvconfig` for local dev use against the Elastic docker registry, that does the necessary setup?

Reply:
Yes it makes sense but I think it's better to keep the two things orthogonal.

You might want to reproduce something that happens in a given job (pr, e2e/master, e2e/stack-versions, etc) using a given registry (elastic/ci, elastic/dev, gcr).